spark_pipeline_framework/transformers/framework_fhir_meta_updater/v2/framework_fhir_meta_updater.py

```diff
@@ -0,0 +1,161 @@
+from typing import Dict, Any, Optional
+
+# noinspection PyProtectedMember
+from spark_pipeline_framework.utilities.capture_parameters import capture_parameters
+from pyspark.ml.param import Param
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import (
+    struct,
+    lit,
+    col,
+    concat,
+    array,
+    expr,
+)
+from spark_pipeline_framework.logger.yarn_logger import get_logger
+from spark_pipeline_framework.progress_logger.progress_logger import ProgressLogger
+from spark_pipeline_framework.transformers.framework_transformer.v1.framework_transformer import (
+    FrameworkTransformer,
+)
+
+
+class FrameworkFhirMetaUpdater(FrameworkTransformer):
+    # noinspection PyUnusedLocal
+    @capture_parameters
+    def __init__(
+        self,
+        *,
+        resource_type: str,
+        source_view: str,
+        view: str,
+        source_url: str,
+        owner_code: str,
+        access_code: Optional[str] = None,
+        vendor_code: Optional[str] = None,
+        source_assigning_authority_code: Optional[str] = None,
+        connection_type_code: Optional[str] = None,
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ):
+        """
+        Used to add/update the meta field with the b.well mandatory stuff
+        :param resource_type: type of resource to update
+        :param source_view: read from this view
+        :param view: write to this view
+        :param source_url: URL for `meta.source`
+        :param owner_code: owner security tag
+        :param access_code: access security tag
+        :param vendor_code: vendor security tag
+        :param source_assigning_authority_code: sourceAssigningAuthority security tag
+        :param connection_type_code: connection_type security tag
+        :param name: Name for the transformer
+        :param parameters: Parameters
+        :param progress_logger: ProgressLogger
+        """
+        super().__init__(
+            name=name, parameters=parameters, progress_logger=progress_logger
+        )
+
+        self.logger = get_logger(__name__)
```

Comment:
I don't think we need this

Reply:
base copied from v1. will check if it can be removed